### ISSUE TITLE
Inconsistent React Error when running webchat using botchat.js

### ISSUE THREAD

**Issue by alokraj68** [microsoft/BotFramework-WebChat#2879]
## Screenshots

![react error](https://user-images.githubusercontent.com/9208731/73742006-70595280-4771-11ea-98a3-64d2a1a2babd.png)

## Version

4.7.1 botchat.js

```
<meta name="botframework-directlinespeech:version" content="4.7.1">
<meta name="botframework-webchat:bundle:variant" content="full">
<meta name="botframework-webchat:bundle:version" content="4.7.1">
<meta name="botframework-webchat:core:version" content="4.7.1">
<meta name="botframework-webchat:ui:version" content="4.7.1">
```

## Describe the bug

We are getting an inconsistent react error as per the screenshot. It points to the URL https://reactjs.org/docs/error-decoder.html/?invariant=321

This further breaks the functionality and chatbot will not work until we reload the page completely. This error is thrown while initiating the chatbot.

This issue was not reproducible in 4.6 version of botchat.js

## Steps to reproduce

1. Go to chatbot page
1. When chatbot does not show any message, check the console.

## Expected behavior

No error should come in console and chatbot should work. Atleast there should be a workaround than reloading the page.

[Bug]

**Comment by stevkan** [comment]
Can you clarify which version of Web Chat you are using? You mention using `botchat.js` which is the v3 version but the meta data you provide indicates you're using 4.7.1 which is v4. Unfortunately, v4 is not backwards compatible if you are trying to use v3 methods of instantiating Web Chat.

You either need to use the botchat.js CDN or migrate your Web Chat code to v4 methods and implementations.

_Please note: If you are using v3 botchat, be aware that it is now deprecated. Instructions for migrating can be found [here](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/MIGRATION.md)._

**Comment by alokraj68** [comment]
Hi, please excuse the typo its. It's 4.7.1. I am using the same as
indicated in the meta data.

Thank you for pointing that out.

**Comment by stevkan** [comment]
A few questions for you:

- Can you tell me which version of React your project is using?
- Have you implemented any custom components, activityHandlers, or middleware that interacts with or relies on Web Chat?
- Are you able to post, or otherwise make accessible, the code you are using (keeping any secrets/tokens hidden)? It would make diagnosing an easier process.

**Comment by alokraj68** [comment]
@stevkan ,

Here are the answers to your questions.

### Can you tell me which version of React your project is using?

We are not using react anywhere in our project.

We are only consuming the webchat.js from CDN.

### Have you implemented any custom components, activityHandlers, or middleware that interacts with or relies on Web Chat?

Yes, We have webspeechponyfillfactory and dispatch stores that relies on webchat.js

### Are you able to post, or otherwise make accessible, the code you are using (keeping any secrets/tokens hidden)? It would make diagnosing an easier process.

Sorry, We won't be able to do that right now.

**Comment by stevkan** [comment]
Thank you for the information.

As I'm unable to reproduce the error, can you test using v4.7.0 (CDN link [here](https://github.com/microsoft/BotFramework-WebChat/releases/tag/v4.7.0))? I'd like to know if the issue is specific to 4.7.1 or not.

**Comment by stevkan** [comment]
Thank you for making the code accessible for testing. As it is, I have been able to run the web chat code successfully and unable to reproduce the error. There are some functions that are called that I had to comment out (i.e. showTyping(), scrollOnNewMessage(), etc.) that I don't have the code for. I'm suspecting one of these may be the source of the error.

I'm hoping I can get the code for these functions, as well. I have emailed Sanket in hope I can get the code in the same manner.
